### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by VladBrynzia**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -37,7 +45,15 @@ function calculateMenAverageAge(people, century) {
...
-  // write code here
+  const womens = people.filter(withChildren ? mother =>
+    people.find(children => children.mother === mother.name) : elem =>
+    elem.sex === 'f');
+
+  const womensAge = womens.map(elem => elem.died - elem.born);
+
+  const averageAgeOfWomens = womensAge.reduce((a, b) => a + b);
+
+  return averageAgeOfWomens / womens.length;
```
Comment (RomanKushyk) [inline src/averageAges.js:56]: You can do it with single .reduce and make as function for reuse

**Review by RomanKushyk — APPROVED** [review]
Very good

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 2]
```diff
@@ -14,12 +14,20 @@
...
+// replace `if ()` statement with &&, || or ?:
+// without nesting
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mens = people.filter(elem => elem.sex === 'm'
+  && century ? Math.ceil(elem.died / 100) === century : elem.sex === 'm');
```
Comment (Ihor-Karpyn) [inline src/averageAges.js:24]:
```suggestion
  const mens = people.filter(person => century 
    ? Math.ceil(person.died / 100) === centur && ?? person.sex === 'm'
    : person.sex === 'm'
 );
```

File: src/averageAges.js [hunk 3]
```diff
@@ -37,7 +45,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womens = people.filter(withChildren ? mother =>
+    people.find(children => children.mother === mother.name) : elem =>
+    elem.sex === 'f');
```
Comment (Ihor-Karpyn) [inline src/averageAges.js:50]:
```suggestion
  const womens = people.filter(withChildren 
    ? mother => people.find(children => children.mother === mother.name)
    : elem => elem.sex === 'f'
  );
```

File: src/averageAges.js [hunk 4]
```diff
@@ -37,7 +45,15 @@ function calculateMenAverageAge(people, century) {
...
+
+  const womensAge = womens.map(elem => elem.died - elem.born);
+
+  const averageAgeOfWomens = womensAge.reduce((a, b) => a + b) / womens.length;
+
+  return averageAgeOfWomens;
 }
```
Comment (Ihor-Karpyn) [inline src/averageAges.js:57]:
```suggestion
  return womensAge.reduce((a, b) => a + b) / womens.length;
}
```

File: src/averageAges.js [hunk 5]
```diff
@@ -55,7 +71,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithSon = people.filter(child => onlyWithSon
+    ? people.find(person => person.name === child.mother
+  && child.sex === 'm')
```
Comment (Ihor-Karpyn) [inline src/averageAges.js:76]:
```suggestion
    ? people.find(person => (
      person.name === child.mother && child.sex === 'm'
    )
```

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**